pkg/util/acrtoken/acrtoken.go

```diff
@@ -98,9 +99,55 @@ func (m *manager) EnsureTokenAndPassword(ctx context.Context, rp *api.RegistryPr
 		return "", err
 	}
 
+	return m.generateTokenPassword(ctx, mgmtcontainerregistry.TokenPasswordNamePassword1, rp)
+}
+
+// RotateTokenPassword chooses either the unused token password or the token
+// password with the oldest creation date, generates a new password, and
+// then updates the registry profile with the newly generated password.
+func (m *manager) RotateTokenPassword(ctx context.Context, rp *api.RegistryProfile) error {
+	tokenProperties, err := m.tokens.GetTokenProperties(ctx, m.r.ResourceGroup, m.r.ResourceName, rp.Username)
+	if err != nil {
+		return err
+	}
+	tokenPasswords := *tokenProperties.Credentials.Passwords
+
+	var passwordToRenew mgmtcontainerregistry.TokenPasswordName
+	switch {
+	// Passwords has zero entries: generate password 1
+	// this shouldn't ever happen, which guarantees it will happen
+	case len(tokenPasswords) == 0:
```

Comment:
Different numbers of passwords on the token get different responses from the API. We could probably figure out a way to use `range` to go through a map of arbitrary length, but as of now the max size is 2 and any change to that would come from Azure and would be a fundamental change to how the container registry tokens work. Since that's pretty unlikely, enumerating the possible sizes explicitly was the path of least resistance.